aws_signing_helper/pkcs11_signer.go

```diff
@@ -77,14 +206,45 @@ fail:
 	return nil, 0, err
 }
 
-// Gets all certificates within the PKCS #11 session
-func getCerts(module *pkcs11.Ctx, session pkcs11.SessionHandle) (certs []*x509.Certificate, err error) {
+// Convert the object-related fields in a URI to pkcs11.Attributes for FindObjectsInit()
+func getFindTemplate(uri *pkcs11uri.Pkcs11URI, class uint) (template []*pkcs11.Attribute) {
+	var v string
+	var ok bool
+
+	template = append(template, pkcs11.NewAttribute(pkcs11.CKA_CLASS, class))
+
+	if uri == nil {
+		return template
+	}
+
+	v, ok = uri.GetPathAttribute("object", false)
+	if ok {
+		template = append(template, pkcs11.NewAttribute(pkcs11.CKA_LABEL, v))
+	}
+	v, ok = uri.GetPathAttribute("id", false)
+	if ok {
+		template = append(template, pkcs11.NewAttribute(pkcs11.CKA_ID, v))
+	}
+	return template
+}
+
+
+// In our list of certs we want to remember the CKA_ID/CKA_LABEL too
+type CertObjInfo struct {
+	id	[]byte
+	label	[]byte
+	x509	*x509.Certificate
+}
+
+// Gets certificate(s) within the PKCS #11 session (i.e. a given token) that
+// match the given uri (and additional criteria in certIdentifier)
+func getCertsInSession(module *pkcs11.Ctx, session pkcs11.SessionHandle, uri *pkcs11uri.Pkcs11URI, certIdentifier CertIdentifier, single bool) (certs []CertObjInfo, err error) {
```

Comment:
Minor, but I don't think `single` is being used here?

Reply:
Hm yeah, I'm fairly sure it *was* at some point. Will check if the failure to use it is a bug, or if it can really die.

Reply:
Killed it in commit 8cb2a2dc92